commit 2b04a66df7796afff59abd8c4d940d7576285b83
Author: Jeff Grafton <jgrafton@google.com>
Date:   Wed Nov 22 12:19:38 2017 -0800

    Rename tree state from 'git archive' to 'archive'

diff --git a/hack/lib/version.sh b/hack/lib/version.sh
index 3dab71052a..0df161e37c 100644
--- a/hack/lib/version.sh
+++ b/hack/lib/version.sh
@@ -22,6 +22,7 @@
 #          source code.
 #    KUBE_GIT_TREE_STATE - "clean" indicates no changes since the git commit id
 #        "dirty" indicates source code changes after the git commit id
+#        "archive" indicates the tree was produced by 'git archive'
 #    KUBE_GIT_VERSION - "vX.Y" used to indicate the last release version.
 #    KUBE_GIT_MAJOR - The major part of the version
 #    KUBE_GIT_MINOR - The minor component of the version
@@ -40,7 +41,7 @@ kube::version::get_version_vars() {
   # we likely don't have a git tree, but these magic values may be filled in.
   if [[ '%' == "%" ]]; then
     KUBE_GIT_COMMIT='6260bb08c46c31eea6cb538b34a9ceb3e406689c'
-    KUBE_GIT_TREE_STATE="git archive"
+    KUBE_GIT_TREE_STATE="archive"
     # When a 'git archive' is exported, the 'tag: v1.8.6' below will look
     # something like 'HEAD -> release-1.8, tag: v1.8.3' where then 'tag: '
     # can be extracted from it.
