This doesn't work. yet.

--- moby-1.13.0/daemon/config.go.orig	2017-08-01 22:16:01.107978983 +1000
+++ moby-1.13.0/daemon/config.go	2017-08-01 22:16:27.524079309 +1000
@@ -98,7 +98,7 @@
 	RawLogs              bool                `json:"raw-logs,omitempty"`
 	Root                 string              `json:"graph,omitempty"`
 	SocketGroup          string              `json:"group,omitempty"`
-	TrustKeyPath         string              `json:"-"`
+	TrustKeyPath         string              `json:"trust-key,omitempty"`
 	CorsHeaders          string              `json:"api-cors-header,omitempty"`
 	EnableCors           bool                `json:"api-enable-cors,omitempty"`
 
@@ -185,6 +185,7 @@
 	flags.StringVar(&config.ClusterAdvertise, "cluster-advertise", "", "Address or interface name to advertise")
 	flags.StringVar(&config.ClusterStore, "cluster-store", "", "URL of the distributed storage backend")
 	flags.Var(opts.NewNamedMapOpts("cluster-store-opts", config.ClusterOpts, nil), "cluster-store-opt", "Set cluster store options")
+	//flags.StringVar(&config.TrustKeyPath, "trust-key", "", "Set legacy trust key path")
 	flags.StringVar(&config.CorsHeaders, "api-cors-header", "", "Set CORS headers in the Engine API")
 	flags.IntVar(&maxConcurrentDownloads, "max-concurrent-downloads", defaultMaxConcurrentDownloads, "Set the max concurrent downloads for each pull")
 	flags.IntVar(&maxConcurrentUploads, "max-concurrent-uploads", defaultMaxConcurrentUploads, "Set the max concurrent uploads for each push")
diff -ur moby-1.13.0.orig/cli/flags/common.go moby-1.13.0/cli/flags/common.go
--- moby-1.13.0.orig/cli/flags/common.go	2017-08-02 22:45:57.073128561 +1000
+++ moby-1.13.0/cli/flags/common.go	2017-08-06 11:14:29.900001488 +1000
@@ -56,6 +56,7 @@
 	flags.StringVarP(&commonOpts.LogLevel, "log-level", "l", "info", "Set the logging level (\"debug\", \"info\", \"warn\", \"error\", \"fatal\")")
 	flags.BoolVar(&commonOpts.TLS, "tls", false, "Use TLS; implied by --tlsverify")
 	flags.BoolVar(&commonOpts.TLSVerify, FlagTLSVerify, dockerTLSVerify, "Use TLS and verify the remote")
+	flags.StringVar(&commonOpts.TrustKey, "trust-key", "", "Legacy trust key path")
 
 	// TODO use flag flags.String("identity"}, "i", "", "Path to libtrust key file")
 
diff -ur moby-1.13.0.orig/cmd/dockerd/daemon.go moby-1.13.0/cmd/dockerd/daemon.go
--- moby-1.13.0.orig/cmd/dockerd/daemon.go	2017-08-02 22:45:57.109128579 +1000
+++ moby-1.13.0/cmd/dockerd/daemon.go	2017-08-06 11:20:35.868217057 +1000
@@ -404,6 +404,7 @@
 	config.TLS = opts.common.TLS
 	config.TLSVerify = opts.common.TLSVerify
 	config.CommonTLSOptions = daemon.CommonTLSOptions{}
+	config.TrustKeyPath = opts.common.TrustKey
 
 	if opts.common.TLSOptions != nil {
 		config.CommonTLSOptions.CAFile = opts.common.TLSOptions.CAFile
